api/handlers_test.go

```diff
@@ -221,3 +267,71 @@ func TestAddRecipeReturnsError(t *testing.T) {
 		So(len(mockedDataStore.AddRecipeCalls()), ShouldEqual, 0)
 	})
 }
+
+func TestUpdateRecipeReturnsOK(t *testing.T) {
+	t.Parallel()
+	Convey("A successful request to update recipe in mongo returns 200 OK response", t, func() {
+		r := httptest.NewRequest("PUT", "http://localhost:22300/recipes/123", bytes.NewBufferString(recipeTest))
+		w := httptest.NewRecorder()
+		mockedDataStore := &storetest.StorerMock{
+			GetRecipeFunc: func(ID string) (*recipe.Response, error) {
+				return &recipe.Response{ID: "123", Alias: "Original", Format: "v3"}, nil
+			},
+			UpdateRecipeFunc: func(ID string, recipeUpdate recipe.Response) error {
+				return nil
+			},
+		}
+		api := GetAPIWithMocks(mockedDataStore)
+		api.Router.ServeHTTP(w, r)
+
+		So(w.Code, ShouldEqual, http.StatusOK)
+		So(len(mockedDataStore.UpdateRecipeCalls()), ShouldEqual, 1)
+		So(len(mockedDataStore.GetRecipeCalls()), ShouldEqual, 2)
```

Comment:
If you have made my suggested changes in above comments this will be reduced to only 1 call

Reply:
Done in commit ""